Sofa/framework/Helper/src/sofa/helper/AdvancedTimer.h

```diff
@@ -216,12 +216,8 @@ class SOFA_HELPER_API AdvancedTimer
             /// return the instance of the factory. Creates it if doesn't exist yet.
             static IdFactory& getInstance()
             {
-                SOFA_THREAD_SPECIFIC_PTR(IdFactory, instance);
-                if (instance == nullptr)
-                {
-                    instance = new IdFactory;
-                }
-                return *instance;
+                static thread_local IdFactory factory;
```

Comment:
> A priori (Francois J.) since C+11 a thread-safe instantiation is done when initializing a `static` variable While here using `thread_local`, it creates one static variable per thread

Yes I agree, but this is just the initial intention. I did not modify any behavior. My goal was just to replace a complex per-thread variable declaration by a portable one. Nothing else.

> AdvancedTimer may not itself be threadsafe

Probably, but it is out of the scope of this pull request.


To be clear: I don't have the ambition to fix anything in `AdvancedTimer`, or change any behavior, in this pull request. Just "decomplexify" the code.